src/movies.js

```diff
@@ -1,16 +1,118 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(array) {
+    let directorsArr = array.map((movie) => {
+        return movie.director
+    })
+    return directorsArr
+}
+console.log(getAllDirectors(movies));
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(array) {
+    let dramaSpielberg = array.filter((movie) => {
+        return (movie.director === "Steven Spielberg") && (movie.genre.includes("Drama"))
+    })
+    return dramaSpielberg.length
+}
+console.log(howManyMovies(movies));
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(array) {
+    if (!array.length) {
+        return 0
+    } else {
+        let allRates = array.reduce((acc, movie) => {
+                if (movie.rate) {
```

Comment:
check indentation here, there seems to be different indentation levels in this function.